etl-pipeline/managers/s3.py

```diff
@@ -27,27 +29,59 @@ def __init__(self):
         )
 
     def store_pdf_manifest(self, record: Record, bucket_name, flags=FileFlags(reader=True), path: str=None):
-        record_id = record.source_id.split('|')[0]
-        pdf_part = next(filter(lambda part: part.file_type == 'application/pdf', record.parts), None)
-
-        if pdf_part is not None:
-            if path:
-                manifest_path = f'manifests/{path}/{record.source}/{record_id}.json'
-            else:
-                manifest_path = f'manifests/{record.source}/{record_id}.json'
-                
-            manifest_url = get_stored_file_url(storage_name=bucket_name, file_path=manifest_path)
-            manifest_json = self.generate_manifest(record=record, source_url=pdf_part.url, manifest_url=manifest_url)
-
-            self.create_manifest_in_s3(manifest_path=manifest_path, manifest_json=manifest_json, bucket=bucket_name)
-
-            record.has_part.insert(0, str(Part(
-                index=pdf_part.index,
-                url=manifest_url,
-                source=record.source,
-                file_type='application/webpub+json',
-                flags=str(flags)
-            )))
+        
+        if record.source == 'doab':
+            link_manager = DOABLinkManager(record)
+
+            link_manager.parse_links()
+
+            print(f'Length of manifests: {len(link_manager.manifests)}')
+            
+            for manifest in link_manager.manifests:
+                manifest_path, manifest_json = manifest
+                self.create_manifest_in_s3(manifest_path=manifest_path, manifest_json=manifest_json, bucket=bucket_name)
+
+                print(f'manifest_path={manifest_path}')
```

Comment:
I recommend pulling this up into the doab mapper to get the right links/manifest into the has_part.